runtime/src/drivers.rs

```diff
@@ -582,6 +582,22 @@ impl Drivers {
         }
     }
 
+    /// Checks if the caller is privilege level 0
+    ///
+    /// # Arguments
+    ///
+    /// * `pl0_pauser` - Value of PL0 PAuser
+    /// * `flags` - Flags from manifest header
+    /// * `locality` - Caller's locality
+    pub fn is_caller_pl0(pl0_pauser: u32, flags: u32, locality: u32) -> bool {
```

Comment:
I think refactoring this and `is_caller_pl1` to return a privilege level enum would be easier to reason about.

Reply:
I think that's reasonable. Feel free to either do that in a this PR or in a follow-up